### ISSUE TITLE
Masonry and Infinite Scroll - overlapping Images (dynamic Gallery on Joomla PHP)

### ISSUE THREAD

**Issue by JohannPiber** [desandro/masonry#1024]
Hello, 
I have a dynamic Photo Gallery with a few thousand images divided into categories plus top lists and a search function (Joomla, PHP, mysqli).
Therefore I have included Masonry style with imagesloaded and infinite scrolling function. All is working fine but sometimes images are overlapping and I need to refresh the page (on Edge, Firefox and Chrome).
Hope someone can help me out - there must me one or more errors I think, since I am quite new to these scripting things. Please explain as to a rookie ;)

Thank you in advance, 
Johann

**PHP Page code:**

    <div id="jgal-masonry">
        <div class="jgal-grid-sizer"></div>
        <?php for($row_count = 0; $row_count < $num_rows; $row_count++ ): ?>
        <?php for($col_count = 0; ($col_count < $this->_config->get('jg_toplistcols')) && ($index < $count); $col_count++):
          $row = $this->rows[$index]; ?>
        <div class="jgal-brick">
            <a href="<?php echo $row->link; ?>"><img src="<?php echo $row->thumb_src; ?>" /></a>
            <div><a href="<?php echo $row->link; ?>"><?php echo ($row->imgtitle; ?></a></div>
        </div>
    <?php $index++; ?>
    <?php endfor ?>
    <?php endfor ?>
    </div>
    <script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>
    <script type="text/javascript" src="/includes/jquery-ias.min.js"></script>
    <script type="text/javascript" src="/includes/imagesloaded.pkgd.min.js"></script>
    <script type="text/javascript" src="/includes/masonry.pkgd.min.js"></script>
    <script type="text/javascript" src="/includes/loadmore.js"></script>

**The CSS:**

    .jgal-masonry{margin:0 auto;padding:0;-moz-column-gap:1.5em;-webkit-column-gap:1.5em;column-gap:1.5em;font-size:.85em;}
    .jgal-brick{display:inline-block;padding:2px;margin:auto;margin-bottom:1px;box-sizing:border-box;-moz-box-sizing:border-box;-webkit-box-sizing:border-box;position:relative;line-height:1em;}
    .jgal-brick{width:24.7%;}
    .jgal-grid-sizer{width:25%;}
    .jgal-grid-sizer:after{content:'';display:block;clear:both;}
    @media only screen and (min-width:400px){.jgal-masonry{-moz-column-count:2;-webkit-column-count:2;column-count:2;}}
    @media only screen and (min-width:700px){.jgal-masonry{-moz-column-count:3;-webkit-column-count:3;column-count:3;}}
    @media only screen and (min-width:900px){.jgal-masonry{-moz-column-count:4;-webkit-column-count:4;column-count:4;}}
    @media only screen and (min-width:1100px){.jgal-masonry{-moz-column-count:4;-webkit-column-count:4;column-count:4;}}
    @media only screen and (min-width:1280px){.jgal-wrapper{width:1260px;}}
    @media screen and (max-width:1224px){.jgal-grid-sizer{width:25.00%;}.jgal-brick{width:24.70%;}}
    @media screen and (max-width:720px){.jgal-grid-sizer{width:33.33%;}.jgal-brick{width:33.33%;}}
    @media screen and (max-width:480px){.jgal-grid-sizer{width:100%;}.jgal-brick{width:100%;}}
    .ias-spinner{position:absolute;bottom:10px;width:100%;text-align:center;margin-left:-16px;}
    .ias-noneleft{text-align:center;color:#ffffff;background-color:#6e9c27;border-radius:5px;position:absolute;width:100%;bottom:-15px;}

**And the Javascript (loadmore.js):**

    var container=document.querySelector('#jgal-masonry');
        jQuery("#jgal-masonry").imagesLoaded(function(){
            var msnry=new Masonry(container,{
				itemSelector:'.jgal-brick',
				columnWidth:'.jgal-grid-sizer',});
    var ias=jQuery.ias({container:'#jgal-masonry',
            item:'.jgal-brick',
            pagination:'.pagination',
            next:'.pagination-next a',
            negativeMargin:30,delay:250,
            isAnimated: true,
            animationOptions: {
            duration: 750,
            queue: false,}
    });
    ias.on('render',function(items){$(items).css({opacity:0});});
    ias.on('rendered',function(items){msnry.appended(items);});
    ias.extension(new IASSpinnerExtension({html:'<div class="ias-spinner" style="text-align: center; font-size:150%; color: #CCCCCC;"><i class="fa fa-spinner fa-pulse fa-3x fa-fw"></i><br />Loading more Images...</div>'}));
    ias.extension(new IASNoneLeftExtension({text:'<div class="ias-noneleft"><em>You reached the end!</em></div>',}));
    });

**Comment by desandro** [comment]
Sorry, without a live URL or a reduced test case, I will not be able to resolve this issue. See [Submitting Issues in the contributing guidelines](https://github.com/desandro/masonry/blob/master/.github/contributing.md#submitting-issues).

**Comment by JohannPiber** [comment]
Sorry, I thought the code would be enough because the Problem seems to be the Paging part of the script. I have now got loose of paging in the Gallery (set paging to 2000 per page) and the grid is working quite fine. The loading time is bad now :(
The search module still has paging function though and sometimes images still overlap.
Thank you anyway, 
Johann
PS: http://www.ironic.site
